File: app/src/main/java/tw/ctl/interest/history/HistoryAdapter.java
package tw.ctl.interest.history;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.realm.RealmResults;
import tw.ctl.interest.Entity;
import tw.ctl.interest.R;

/**
 * Created by jacklai on 2017/3/1.
 */

public class HistoryAdapter extends RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder> {

    private RealmResults<Entity> entities;

    @Override
    public HistoryViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_entity, parent, false);
        return new HistoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(HistoryViewHolder holder, int position) {
        Entity entity = entities.get(position);
        holder.simpleResult.setText(entity.simpleResult);
        holder.compoundResult.setText(entity.compoundResult);
        holder.investResult.setText(entity.investResult);
        holder.principal.setText(entity.principal);
        holder.interest.setText(entity.interest);
        holder.period.setText(entity.period);
        holder.invest.setText(entity.invest);
    }

    @Override
    public int getItemCount() {
        return entities == null ? 0 : entities.size();
    }

    public void setEntities(RealmResults<Entity> entities) {
        this.entities = entities;
        notifyDataSetChanged();
    }

    class HistoryViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.simple_interest_result)   TextView simpleResult;
        @BindView(R.id.compound_interest_result) TextView compoundResult;
        @BindView(R.id.invest_interest_result)   TextView investResult;
        @BindView(R.id.principal) TextView principal;
        @BindView(R.id.interest)  TextView interest;
        @BindView(R.id.period)    TextView period;
        @BindView(R.id.invest)    TextView invest;

        public HistoryViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

    }

}

File: app/src/main/java/tw/ctl/interest/history/HistoryAdapter.kt
package tw.ctl.interest.history

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.item_record.view.*
import tw.ctl.interest.R
import tw.ctl.interest.model.Record

class HistoryAdapter : RecyclerView.Adapter<HistoryAdapter.HistoryViewHolder>() {

    private var records = listOf<Record>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HistoryViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_record, parent, false)
        return HistoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: HistoryViewHolder, position: Int) {
        holder.bindRecord(records[position])
    }

    override fun getItemCount(): Int = records.size

    fun setRecords(records: List<Record>) {
        this.records = records
        notifyDataSetChanged()
    }

    class HistoryViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        fun bindRecord(record: Record) {
            with(itemView) {
                simpleInterest.text = record.simpleResult
                compoundInterest.text = record.compoundResult
                investInterest.text = record.investResult
                principal.text = record.principal
                interest.text = record.interest
                periodLabel.text = context.getString(R.string.period_with_type, "(${record.periodType})")
                period.text = record.period
                investLabel.text = context.getString(R.string.invest, record.periodType, "")
                invest.text = record.invest
            }
        }
    }

}
